src/charm.py

```diff
@@ -38,11 +43,25 @@ def __init__(self, *args: typing.Any):
         self.traefik_route = traefik_route_observer.TraefikRouteObserver(self)
         self.opensearch = opensearch_observer.OpenSearchObserver(self)
 
+        self.framework.observe(self.on.install, self._on_install)
         self.framework.observe(
             self.on.wazuh_server_pebble_ready, self._on_wazuh_server_pebble_ready
         )
         self.framework.observe(self.on.config_changed, self._on_config_changed)
 
+    def _on_install(self, _: ops.InstallEvent) -> None:
+        """Install event handler."""
+        if self.unit.is_leader():
+            try:
+                self.model.get_secret(label=WAZUH_CLUSTER_KEY_SECRET_LABEL)
+            except ops.SecretNotFoundError:
+                logger.debug(
+                    "Secret with label %s not found. Creating one.", WAZUH_CLUSTER_KEY_SECRET_LABEL
+                )
+                self.unit.add_secret(
+                    {"value": secrets.token_hex(8)}, label=WAZUH_CLUSTER_KEY_SECRET_LABEL
+                )
```

Comment:
What are the Wazuh's recommendations on secret rotation?
Shouldn't the secret have an expiration date (or a rotation policy), and should a `secret_expired` or `secret_rotate` hook be added?

Reply:
Also shouldn't a grant be added, so that all unit get access to it ? And the secret_id added to the peer relation.

Reply:
I don't need the secret id to be added to the relation since I'm fetching it by label, do I? Also, iirc the secrets are app scoped

Reply:
Well the secret here is created on the unit, and not on the app, if you want it to be app scoped, then you need to use `self.app`
This should be tested and covered with an integration test

Reply:
Concerning the label, the documentation seems to agree with you: `The secret ID is normally only used to exchange a reference to the secret between applications. Within a single application, all you need is the secret label.`

Reply:
What about secret rotation / expiration ?

Reply:
I don't see any recommendation in terms of secret rotation, do you have any suggestion?

Reply:
I don't see any either, the doc only explains it's an encryption key used for servers in the cluster to communicate.
We may want to ask Wazuh if they have any recommendations, we can also ask Stephanie what their preference 
 is there.
For now, let's keep it fixed after a random generation.